bluemira/geometry/imprint_solids.py

```diff
@@ -0,0 +1,311 @@
+# SPDX-FileCopyrightText: 2021-present M. Coleman, J. Cook, F. Franza
+# SPDX-FileCopyrightText: 2021-present I.A. Maione, S. McIntosh
+# SPDX-FileCopyrightText: 2021-present J. Morris, D. Short
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+"""
+Imprinting solids together.
+"""
+
+from collections.abc import Iterable
+from typing import Any
+
+from rich.progress import track
+
+from bluemira.base.look_and_feel import bluemira_print
+from bluemira.geometry.overlap_checking import find_approx_overlapping_pairs
+from bluemira.geometry.solid import BluemiraSolid
+
+try:
+    from OCC.Core.BOPAlgo import BOPAlgo_MakeConnected
```

Comment:
Maybe a note about the use of pythonocc here
```suggestion
    # TODO @oliverfunk: pythonocc exposes functionality that cant be accessed 
    # through the freecad API. Should it become available this 
    # should be moved to freecadapi
    from OCC.Core.BOPAlgo import BOPAlgo_MakeConnected
```